Running global `jest-cli` should warn instead of error

The [`process.exit(1)` call](https://github.com/facebook/jest/blob/a737c6/packages/jest-cli/src/cli/getJest.js#L36-L40) triggered when running a globally-installed version of Jest results in odd test results - even when all tests pass, the process returns a non-zero exit code, and it's not immediately clear why.

When `jest` is assigned to `npm test` (which is a pretty common convention), you get:

```
Please run `npm install` to use the version of Jest intended for this project.
...
163 tests passed (163 total in 30 test suites, run time 3.292s)
npm ERR! Test failed.  See above for more details.
```

There are no obvious details (the "Please run..." line doesn't look like an error message") and no failing tests, so debugging is awkward.

**Recommendations**

1. I would prefer to remove the test for a package-local version of Jest from `getJest` entirely. If people want to use a non-local version of Jest without running `npm install`, I'm not sure why the package should care?
2. If the test is useful enough to remain, it should use `console.warn` without affecting the exit code because the tests may still run without trouble, and the warning is sufficient indication that the output might not reflect exactly what the user intended.
3. If returning a non-zero exit code is preferred, the test should throw a real `Error` to make debugging easier -  Node's built-in error handling is much more explicit than `console.error` followed by `process.exit(1)`.
4. If you really don't want to throw an error, call `process.exit(1)` immediately rather than waiting for all the tests to run - if it's not okay to run Jest globally when there's a missing locally-defined dependency, then it shouldn't run at all.

--- a/packages/jest-cli/src/cli/getJest.js
+++ b/packages/jest-cli/src/cli/getJest.js
@@ -12,6 +12,7 @@
 
 import type {Path} from 'types/Config';
 
+const chalk = require('chalk');
 const fs = require('graceful-fs');
 const path = require('path');
 
@@ -33,11 +34,12 @@ function getJest(packageRoot: Path) {
         (dependencies && dependencies['jest-cli']) ||
         (devDependencies && devDependencies['jest-cli'])
       ) {
-        console.error(
-          'Please run `npm install` to use the version of Jest intended for ' +
-          'this project.',
-        );
-        process.on('exit', () => process.exit(1));
+        process.on('exit', () => console.log(
+          chalk.red(
+            'Please run `npm install` to use the version of Jest intended ' +
+            'for this project.',
+          ),
+        ));
       }
     }
     return jest;

